haystack/document_store/elasticsearch.py

```diff
@@ -699,10 +699,36 @@ def add_eval_data(self, filename: str, doc_index: str = "eval_document", label_i
         :type label_index: str
         """
 
-        docs, labels = eval_data_from_file(filename)
+        docs, labels = eval_data_from_json(filename)
```

Comment:
Why is one method using json and the other jsonl? Because of the batching via a generator? If yes, it would probably be cleaner again to handle this under the hood, possibly via a param `filetype` in `add_eval_data()`. If batch_size != None and filetype = json, we could convert to jsonl before going on.